pkg/deploy/generator/resources_rp.go

```diff
@@ -37,98 +37,132 @@ func (g *generator) rpManagedIdentity() *arm.Resource {
 	}
 }
 
-func (g *generator) rpSecurityGroup() *arm.Resource {
-	rules := []mgmtnetwork.SecurityRule{
-		{
-			SecurityRulePropertiesFormat: &mgmtnetwork.SecurityRulePropertiesFormat{
-				Protocol:                 mgmtnetwork.SecurityRuleProtocolTCP,
-				SourcePortRange:          to.StringPtr("*"),
-				DestinationPortRange:     to.StringPtr("443"),
-				SourceAddressPrefix:      to.StringPtr("AzureResourceManager"),
-				DestinationAddressPrefix: to.StringPtr("*"),
-				Access:                   mgmtnetwork.SecurityRuleAccessAllow,
-				Priority:                 to.Int32Ptr(120),
-				Direction:                mgmtnetwork.SecurityRuleDirectionInbound,
-			},
-			Name: to.StringPtr("rp_in_arm"),
-		},
-		{
-			SecurityRulePropertiesFormat: &mgmtnetwork.SecurityRulePropertiesFormat{
-				Protocol:                 mgmtnetwork.SecurityRuleProtocolTCP,
-				SourcePortRange:          to.StringPtr("*"),
-				DestinationPortRange:     to.StringPtr("443"),
-				SourceAddressPrefix:      to.StringPtr("GenevaActions"),
-				DestinationAddressPrefix: to.StringPtr("*"),
-				Access:                   mgmtnetwork.SecurityRuleAccessAllow,
-				Priority:                 to.Int32Ptr(130),
-				Direction:                mgmtnetwork.SecurityRuleDirectionInbound,
-			},
-			Name: to.StringPtr("rp_in_geneva"),
-		},
-	}
+func (g *generator) rpSecurityGroup() []*arm.Resource {
```

Comment:
This function has been changed to be too complicated to achieve the conditional adding of a single security rule. Duplicate rp-nsg blocks in the ARM template is also not optimal.

Please can you rather add a new function `(func (g *generator) rpSecurityGroupFairFaxAdminPortal() *arm.Resource` that effectively adds this ARM resource:
```
{
    "type": "Microsoft.Network/networkSecurityGroups/securityRules",
    "apiVersion": "2022-07-01",
    "name": "rp-nsg/portal_in",
    "condition": "[not(empty(parameters('rpNsgPortalSourceAddressPrefixes')))]",
    "properties": {
        "protocol": "Tcp",
        "sourcePortRange": "*",
        "destinationPortRange": "444",
        "sourceAddressPrefixes": "[parameters('rpNsgPortalSourceAddressPrefixes')]",
        "destinationAddressPrefix": "*",
        "access": "Allow",
        "priority": 142,
        "direction": "Inbound"
    }
}
```` 
That way we are cleanly adding the security rule when the RP-Config array is set with no changes to the existing NSG deployment, i.e. in FairFax.

Reply:
I like this approach much better, I'll give it a shot this way.